### PR TITLE
[ECO-1886] Various chart updates: color scheme, websocket data, fixes

### PR DESCRIPTION
**Pull request by xbtmatt**
# Description

- Update the color scheme to pink and green
- Incorporate the live websockets data into the chart so the bars update in real time
- No re-render on props change (every 2-3 seconds the chart would fully reload- disappearing and then reappearing. This is now fixed by using `useMemo` and avoiding recreating objects with prop data causing re-renders.
- Fix `time order violation` issues (it's an issue with `has_empty_bars`)
- Fix candlesticks not updating upon new candle issues
  - Fixed by adding logic to incoming swap/periodic state events and always fetching the latest candlestick from the fullnode whenever `getBars` is called to get the most recent data.
- Fixed the issue with the chart not rendering once it's done loading
  - Fixed by changing `onLoad` to `onReady` which also fires any time the component is mounted, not just when the script is loaded, see: https://nextjs.org/docs/app/building-your-application/optimizing/scripts#executing-additional-code
- Made candlesticks contiguous by having each incoming `latestBar` update check for the last bar's close value and use it if it exists. For freshly fetched data that isn't stored in state (like in `getBars`) we iterate over each value in state and use the previous bar's close as the new one's open when constructing the `bars` array.
- Add the existing market symbol data with emojis, market ID, etc, to the initial application state. Pass it to the `Chart` so we can search for emojis/symbols in markets and select them inside the `Chart` container with `searchSymbol` and `resolveSymbol`
- Fixed the little ghost showing `No data` for markets with no trading activity yet by adding an initial candlestick bar if there's literally 0 trading activity so far

# Testing

Manual with some new unit tests for a few logical functions

# Checklist

- [x] Did you update relevant documentation?
- [x] Did you add tests to cover new code or a fixed issue?
- [x] Did you update the changelog?
- [x] Did you check all checkboxes from the linked Linear task?

### CODE REVIEW

File: src/typescript/sdk/src/const.ts [hunk 1]
```diff
@@ -46,7 +46,15 @@ export const DECIMALS = 8;
 export const MAX_CHAT_MESSAGE_LENGTH = 100;
 export const MAX_SYMBOL_LENGTH = 10;
 
+/**
+ * Note that a period boundary, a candlestick resolution, a period, and a candlestick timeframe
+ * are all referred to interchangeably throughout this
+ */
 export enum CandlestickResolution {
+  PERIOD_1S = 1000000,
```
Comment (xbtmatt) [inline src/typescript/sdk/src/const.ts:54]: Remove before merge again.

File: src/typescript/sdk/src/const.ts [hunk 2]
```diff
@@ -46,6 +46,10 @@ export const DECIMALS = 8;
 export const MAX_CHAT_MESSAGE_LENGTH = 100;
 export const MAX_SYMBOL_LENGTH = 10;
 
+/**
+ * Note that a period boundary, a candlestick resolution, a period, and a candlestick time frame
+ * are all referred to interchangeably throughout this
```
Comment (xbtmatt) [inline src/typescript/sdk/src/const.ts:51]:
```suggestion
 * are all referred to interchangeably throughout this codebase.
```

**Review by xbtmatt — COMMENTED** [review]
Resolve issues with time violations and switching candlestick time frames.

File: src/typescript/frontend/src/components/charts/README.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
+<!---cspell:words OHLCV -->
```
Comment (alnoki) [inline src/typescript/frontend/src/components/charts/README.md:1]:
```suggestion
<!---cspell:word OHLCV -->
```

File: src/typescript/frontend/src/components/charts/README.md
Comment (alnoki) [inline src/typescript/frontend/src/components/charts/README.md:1]: - [ ] Link to this from a higher-level README, stating that this document captures a lot of the data flow architecture

**Review by alnoki — CHANGES_REQUESTED** [review]
- [ ] Resolve `take out chart search option` from Linear or move to a separate Linear task

File: src/typescript/frontend/src/components/charts/README.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+   update the existing latest bar.
+1. When a new swap event comes in and it should be part of a new candlestick
+   bar, we create a new bar based off of the swap data.
+
+Once we've deserialized the stream buffer data into proper contract events, we
+store them in state and visually update the latest bar data with the
+[subscribeBars] callback function provided by the datafeed API.
+
+We store this callback function in our [EventStore state] and call it whenever
+we receive an event in state _and_ the chart is subscribed to a market..
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/charts/README.md:120]:
```suggestion
we receive an event in state _and_ the chart is subscribed to a market.
```

File: src/typescript/frontend/src/components/charts/README.md
Comment (xbtmatt) [reply src/typescript/frontend/src/components/charts/README.md:1]: Done! added it to `frontend/README.md`

**Review by alnoki — APPROVED** [review]

**Pull request merged**